test/cpu.cpp

```diff
@@ -1150,3 +1151,763 @@ TEST_CASE("TCD", "[cpu][instruction]") {
+				AddressingMode::Immediate,
+				AddressingMode::Absolute,
+			},
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.Y.forceStoreFull(lhs);
+				cpu.setFlag(CPU::flags::x, indexRegistersAre8Bit);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(result >= 0);
+				REQUIRE(cpu.getFlag(CPU::flags::z) == resultIsZero);
+				REQUIRE(cpu.getFlag(CPU::flags::n) == resultIsNegative);
+				REQUIRE(cpu.getFlag(CPU::flags::c) == resultHasCarry);
+			}
+		);
+	}
+}
+
+TEST_CASE("BIT", "[cpu][instruction]") {
+	auto memoryAndAccumulatorAre8Bit = GENERATE(false, true);
+	auto lhs = static_cast<Word>(GENERATE_COPY(take(1, random<Address>(0, (memoryAndAccumulatorAre8Bit ? 0xff : 0xffff) + 1))));
+	auto rhs = static_cast<Word>(GENERATE_COPY(take(1, random<Address>(0, (memoryAndAccumulatorAre8Bit ? 0xff : 0xffff) + 1))));
+	auto result = lhs & rhs;
+	bool resultIsZero = result == 0;
+	bool resultIsNegative = msb(rhs, memoryAndAccumulatorAre8Bit);
+	bool resultHasOverflow = memoryAndAccumulatorAre8Bit ? ((rhs & (1u << 6)) != 0) : ((rhs & (1u << 14)) != 0);
+	
+	DYNAMIC_SECTION((memoryAndAccumulatorAre8Bit ? 8 : 16) << "-bit; ") {
+		testInstructionWithOperand(Opcode::BIT, memoryAndAccumulatorAre8Bit ? 8 : 16, rhs,
+			{
+				AddressingMode::Direct,
+				AddressingMode::Immediate,
+				AddressingMode::Absolute,
+				AddressingMode::DirectIndexedX,
+				AddressingMode::AbsoluteIndexedX,
+			},
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.A.forceStoreFull(lhs);
+				cpu.setFlag(CPU::flags::m, memoryAndAccumulatorAre8Bit);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::z) == resultIsZero);
+				REQUIRE(cpu.getFlag(CPU::flags::n) == resultIsNegative);
+				REQUIRE(cpu.getFlag(CPU::flags::v) == resultHasOverflow);
+			}
+		);
+	}
+}
+
+TEST_CASE("INC", "[cpu][instruction]") {
+	auto memoryAndAccumulatorAre8Bit = GENERATE(false, true);
+	auto lhs = static_cast<Word>(GENERATE_COPY(take(1, random<Address>(0, (memoryAndAccumulatorAre8Bit ? 0xff : 0xffff) + 1))));
+	auto rhs = static_cast<Word>(GENERATE_COPY(take(1, random<Address>(0, (memoryAndAccumulatorAre8Bit ? 0xff : 0xffff) + 1))));
+	auto result = lhs + 1;
+	bool resultIsZero = result == 0;
+	bool resultIsNegative = msb(result, memoryAndAccumulatorAre8Bit);
+	
+	DYNAMIC_SECTION((memoryAndAccumulatorAre8Bit ? 8 : 16) << "-bit; ") {
+		testInstructionWithOperand(Opcode::INC, memoryAndAccumulatorAre8Bit ? 8 : 16, rhs,
+			{
+				AddressingMode::Direct,
+				AddressingMode::Accumulator,
+				AddressingMode::Absolute,
+				AddressingMode::DirectIndexedX,
+				AddressingMode::AbsoluteIndexedX,
+			},
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.A.forceStoreFull(lhs);
+				cpu.setFlag(CPU::flags::m, memoryAndAccumulatorAre8Bit);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.A.load() == result);
```

Comment:
This instruction actually writes back to wherever it loaded the value from, so depending on the addressing mode, it might write back to the accumulator or memory. I'll need to add some code to `testInstructionWithOperand` to handle this case and provided the written value to the test lambda. For now, you can mark this test case as "known to fail" by adding the `[!mayfail]` tag to the `TEST_CASE` definition above.